mystery-deckbuilder/Assets/Scripts/Encounter/EncounterScript.cs

```diff
@@ -90,39 +151,270 @@ public int GetCompliance()
     {
         return _compliance;
     }
-
+    /**
+     * compliance value cannot go bellow 0
+     */
     public void IncCompliance(int inc)
     {
-        _compliance += inc;
+        if (_compliance + inc >= 0)
+        {
+            _compliance += inc;
+        }
+        else
+        {
+            _compliance = 0;
+        }
+        complianceBar.GetComponent<BarScript>().SetValue(_compliance);
         complianceBar.GetComponent<BarScript>().SetValue(_compliance);
         if (_compliance >= _threshold)
         {
             EndEncounter(true);
         }
     }
 
+    public string GetWeakness()
+    {
+        return _weakness;
+    }
+
+    public string GetResistance()
+    {
+        return _resistance;
+    }
+
     /**
-     * compliance value cannot go bellow 0
+     * checks if there is a card to draw
+     * finds a place for the card
+     * updates backend
+     * creates card object
+     * intsatiates prefab
+     * sets prefab to match card object
      */
-    public void DecCompliance(int dec)
+    public void DrawCard()
     {
-        if (_compliance - dec >= 0)
+        if (deck.Count == 0) //if you are out of cards, you can't draw more
         {
-            _compliance -= dec;
+            return;
         }
-        else
+
+        Transform place = null;
+        
+        foreach (KeyValuePair<Transform, bool> i in locations)
         {
-            _compliance = 0;
+            //Debug.Log("for");
+            if (!i.Value)
+            {
+                //Debug.Log("if");
+                place = i.Key;
+                break;
+            }
+        }
+        
+        if (place == null) //if there is no space for new cards they can't draw (hopefully)
+        {
+            return;
+        }
+
+        locations[place] = true;
+
+        IncPatience(-1); //draw patience cost, happens after card draw is successful
```

Comment:
NIT: haha, the purpose of adding DecPatience was that you could decrement by (1) instead of adding (-1)